### PR TITLE
Feature/11468 update caching

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
- added a new caching policy `.loadOnlyOnceADay` to the cached rest service client.
- extended CachedData to make sure we use the client date for storeing and comparing (discussed with @mlenkeit)
- added test to make sure old cached data load correctly
- added unit tests to verify `.loadOnlyOnceADay` beahviour

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11468

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClient/Service/Services/CachedRestService.swift [hunk 1]
```diff
@@ -131,4 +163,6 @@ class CachedRestService: Service {
 
 	private let optionalSession: URLSession?
 	private var cache: KeyValueCaching
+	private var fakeClientCacheDate: Date?
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/HTTPClient/Service/Services/CachedRestService.swift:166]: Hm this name is confusing me and somehow I don't quite understand what its used for.
Somehow it gets initialised from the init that is only available in the fake but then it gets checked from the production code. Feels kind a odd.

File: src/xcode/ENA/ENA/Source/HTTPClient/Service/__tests__/CachedRestServiceTests.swift [hunk 2]
```diff
@@ -224,296 +224,4 @@ class CachedRestServiceTests: XCTestCase {
 		waitForExpectations(timeout: .short)
 	}
 	
-	// MARK: - Cache Policy tests
-	
-	func test_GIVEN_CachePolicyNoNetwork_WHEN_DefaultValueNoCacheIsSet_THEN_DefaultValueIsReturned() {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/HTTPClient/Service/__tests__/CachedRestServiceTests.swift:229]: Are these tests moved somewhere else?

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import Foundation
+
+struct CacheData: Codable {
+
+	// MARK: - Init
+
+	init(
+		data: Data,
+		eTag: String,
+		serverDate: Date?,
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift:14]: serverDate is written but I don't see where it is read. Do we need it?

File: src/xcode/ENA/ENA/Source/HTTPClient/Service/__tests__/CachedRestServiceTests.swift (hunk 2 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/HTTPClient/Service/__tests__/CachedRestServiceTests.swift:229]: yes moved all of them to the new file `CachePolicyTests`

File: src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		self.serverDate = serverDate
+		self.clientDate = clientDate
+	}
+
+	// MARK: - Protocol Codable
+
+	enum CodingKeys: String, CodingKey {
+		case data
+		case eTag
+		case serverDate = "date"
```
[the remark this answers is not in the document]
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift:30]: I added a comment

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		date: Date
+	) {
+		self.data = data
+		self.eTag = eTag
+		self.date = date
+	}
+
+	// MARK: - Protocol Codable
+
+	enum CodingKeys: String, CodingKey {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Workers/Store/CacheData.swift:23]: I think CodingKeys are not needed anymore.

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**